### PR TITLE
Yask backend now capable of jitting and executing

### PR DESCRIPTION
**Pull request by FabioLuporini**
Depends on (and currently includes!)  #329

### CODE REVIEW

File: devito/interfaces.py [hunk 1]
```diff
@@ -357,6 +362,17 @@ class SymbolicData(AbstractSymbol):
...
+    def _data_buffer(self):
+        """Reference to the actual data. This is *not* a view of the data.
+        This method is for internal use only."""
+        return self.data
+
+    @abc.abstractproperty
+    def data(self):
+        """The value of the data object."""
+        return
+
```
Comment (navjotk) [inline devito/interfaces.py]: What is going on here?

Comment (FabioLuporini) [reply devito/interfaces.py]: Python's abstract methods are supposed to return None (see for example https://stackoverflow.com/questions/4382945/abstract-methods-in-python)

This ensures that all derived (concrete) classes will have an implementation for ``data``

**Comment by FabioLuporini** [comment]
all clear @navjotk ?

**Comment by FabioLuporini** [comment]
Merging (offline discussion cleared this out)...

**Comment by FabioLuporini** [comment]
Merged.

**Pull request merged**